File: salesflow-app/src/backend/app/api/schemas/sales_intelligence.py
"""
╔════════════════════════════════════════════════════════════════════════════╗
║  SALES INTELLIGENCE API SCHEMAS v3.0                                       ║
║  Schemas für Multi-Language, Buyer Psychology, Frameworks, Industries     ║
╚════════════════════════════════════════════════════════════════════════════╝
"""

from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any, Literal
from datetime import datetime
from enum import Enum


# =============================================================================
# ENUMS
# =============================================================================

class LanguageCode(str, Enum):
    DE = "de"
    DE_AT = "de-at"
    DE_CH = "de-ch"
    EN_US = "en-us"
    EN_UK = "en-uk"
    ES = "es"
    ES_LATAM = "es-latam"
    FR = "fr"
    IT = "it"
    PT = "pt"
    NL = "nl"
    PL = "pl"
    TR = "tr"


class BuyerType(str, Enum):
    ANALYTICAL = "analytical"
    DRIVER = "driver"
    EXPRESSIVE = "expressive"
    AMIABLE = "amiable"


class BuyingStage(str, Enum):
    AWARENESS = "awareness"
    CONSIDERATION = "consideration"
    DECISION = "decision"
    VALIDATION = "validation"


class RiskProfile(str, Enum):
    RISK_AVERSE = "risk_averse"
    RISK_NEUTRAL = "risk_neutral"
    RISK_TAKER = "risk_taker"


class AuthorityLevel(str, Enum):
    DECISION_MAKER = "decision_maker"
    INFLUENCER = "influencer"
    GATEKEEPER = "gatekeeper"
    CHAMPION = "champion"
    USER = "user"


class FrameworkType(str, Enum):
    SPIN = "spin"
    CHALLENGER = "challenger"
    GAP = "gap"
    SANDLER = "sandler"
    SNAP = "snap"
    MEDDIC = "meddic"
    SOLUTION = "solution"


class IndustryType(str, Enum):
    NETWORK_MARKETING = "network_marketing"
    REAL_ESTATE = "real_estate"
    INSURANCE = "insurance"
    FINANCE = "finance"
    B2B_SAAS = "b2b_saas"
    B2B_SERVICES = "b2b_services"
    COACHING = "coaching"
    AUTOMOTIVE = "automotive"
    RECRUITING = "recruiting"
    HEALTHCARE = "healthcare"
    EVENT_SALES = "event_sales"
    RETAIL_HIGH_TICKET = "retail_high_ticket"


class MomentumTrend(str, Enum):
    IMPROVING = "improving"
    STABLE = "stable"
    DECLINING = "declining"


# =============================================================================
# LANGUAGE DETECTION
# =============================================================================

class LanguageDetectionRequest(BaseModel):
    """Request für Spracherkennung"""
    text: str = Field(..., description="Text zur Analyse")


class LanguageDetectionResponse(BaseModel):
    """Response der Spracherkennung"""
    language_code: str
    regional_variant: Optional[str] = None
    formality_detected: str
    dialect_markers: List[str] = []
    confidence: float


class CulturalProfileResponse(BaseModel):
    """Kulturelles Profil für eine Sprache"""
    language_code: str
    language_name: str
    formality_default: str
    directness: str
    relationship_first: bool
    urgency_acceptable: bool
    small_talk_expected: bool
    emoji_tolerance: str
    typical_objection_style: str
    trust_building_approach: str
    closing_style: str


class LocalizedTemplateRequest(BaseModel):
    """Request für lokalisiertes Template"""
    template_key: str = Field(..., description="z.B. 'follow_up', 'price_objection', 'ghost_buster'")
    language_code: str = Field(default="de")
    lead_name: str = Field(default="[Name]")


class LocalizedTemplateResponse(BaseModel):
    """Response mit lokalisiertem Template"""
    template_key: str
    language_code: str
    template_text: str


# =============================================================================
# BUYER PSYCHOLOGY
# =============================================================================

class BuyerProfileRequest(BaseModel):
    """Request für Buyer Profile Analysis"""
    chat_text: str = Field(..., description="Chat-Verlauf zur Analyse")
    lead_id: Optional[str] = None
    context: Optional[Dict[str, Any]] = None


class BuyerProfileAnalysis(BaseModel):
    """Ergebnis der Buyer Profile Analyse"""
    buyer_type: BuyerType
    buyer_type_confidence: float
    buyer_type_signals: List[str]
    
    buying_stage: BuyingStage
    buying_stage_confidence: float
    buying_stage_signals: List[str]
    
    risk_profile: RiskProfile
    risk_profile_confidence: float
    risk_profile_signals: List[str]
    
    authority_level: AuthorityLevel
    authority_level_confidence: float
    authority_level_signals: List[str]


class CommunicationRecommendation(BaseModel):
    """Empfehlungen für Kommunikation basierend auf Buyer Profile"""
    tone: str
    message_length: str
    emphasis: List[str]
    avoid: List[str]
    ideal_next_message: str


class BuyerProfileResponse(BaseModel):
    """Vollständige Buyer Profile Response"""
    buyer_profile: BuyerProfileAnalysis
    communication_recommendations: CommunicationRecommendation
    objection_prediction: Dict[str, Any]
    adapted_prompt_section: str


class BuyerTypeInfoResponse(BaseModel):
    """Info zu einem Buyer Type"""
    buyer_type: BuyerType
    name: str
    emoji: str
    characteristics: List[str]
    communication_do: List[str]
    communication_dont: List[str]
    ideal_pitch: str
    objection_handling: str
    closing_approach: str
    typical_questions: List[str]


class ObjectionByBuyerTypeRequest(BaseModel):
    """Request für Einwandbehandlung nach Buyer Type"""
    buyer_type: BuyerType
    objection_type: str = Field(..., description="z.B. 'price', 'time', 'trust'")


class ObjectionByBuyerTypeResponse(BaseModel):
    """Response für Einwandbehandlung nach Buyer Type"""
    buyer_type: BuyerType
    objection_type: str
    strategy: str
    example: str


# =============================================================================
# SALES FRAMEWORKS
# =============================================================================

class FrameworkRecommendationRequest(BaseModel):
    """Request für Framework-Empfehlung"""
    deal_size: Optional[str] = Field(None, description="'small', 'medium', 'large', 'enterprise'")
    sales_cycle: Optional[str] = Field(None, description="'short', 'medium', 'long'")
    lead_type: Optional[str] = Field(None, description="'cold', 'warm', 'hot'")
    situation: Optional[str] = Field(None, description="Beschreibung der Situation")


class FrameworkRecommendationResponse(BaseModel):
    """Response mit Framework-Empfehlung"""
    recommended_framework: FrameworkType
    reasoning: str
    alternatives: List[FrameworkType]


class FrameworkInfoResponse(BaseModel):
    """Info zu einem Sales Framework"""
    id: str
    name: str
    best_for: List[str]
    core_principle: str
    stages: List[Dict[str, Any]]
    key_questions: List[str]
    common_mistakes: List[str]


class FrameworkPromptRequest(BaseModel):
    """Request für Framework-spezifischen Prompt"""
    framework_id: FrameworkType
    current_stage: Optional[str] = None
    lead_context: Optional[Dict[str, Any]] = None


class FrameworkPromptResponse(BaseModel):
    """Response mit Framework-spezifischem Prompt"""
    framework_id: str
    framework_name: str
    prompt_section: str
    stage_questions: List[str]


class ObjectionByFrameworkRequest(BaseModel):
    """Request für Einwandbehandlung nach Framework"""
    framework_id: FrameworkType
    objection_type: str = Field(..., description="z.B. 'price', 'time', 'think'")


class ObjectionByFrameworkResponse(BaseModel):
    """Response für Einwandbehandlung nach Framework"""
    framework_id: str
    objection_type: str
    response: str


# =============================================================================
# INDUSTRIES
# =============================================================================

class IndustryInfoResponse(BaseModel):
    """Info zu einer Branche"""
    id: str
    name: str
    description: str
    typical_sales_cycle: str
    avg_deal_size: str
    key_decision_factors: List[str]
    typical_objections: List[str]
    compliance_rules: List[str]
    recommended_frameworks: List[str]
    buyer_personas: List[Dict[str, str]]
    communication_style: Dict[str, str]
    trust_builders: List[str]
    red_flags: List[str]


class IndustryListItem(BaseModel):
    """Eintrag in der Branchen-Liste"""
    id: str
    name: str
    description: str


class IndustryPromptResponse(BaseModel):
    """Response mit branchenspezifischem Prompt"""
    industry_id: str
    industry_name: str
    prompt_section: str


class IndustryObjectionRequest(BaseModel):
    """Request für branchenspezifische Einwandbehandlung"""
    industry_id: IndustryType
    objection_type: str


class IndustryObjectionResponse(BaseModel):
    """Response für branchenspezifische Einwandbehandlung"""
    industry_id: str
    objection_type: str
    strategy: str
    example: str


# =============================================================================
# DEAL MOMENTUM
# =============================================================================

class MomentumSignalInput(BaseModel):
    """Ein Momentum-Signal"""
    type: Literal["positive", "negative", "neutral"]
    signal: str
    weight: float = Field(default=1.0, ge=0.1, le=2.0)
    description: str


class MomentumCalculationRequest(BaseModel):
    """Request für Momentum-Berechnung"""
    lead_id: str
    signals: List[MomentumSignalInput]


class MomentumScoreResponse(BaseModel):
    """Response mit Momentum Score"""
    lead_id: str
    score: float = Field(..., ge=1, le=10)
    trend: MomentumTrend
    recommendation: str
    signals_count: int
    positive_signals: int
    negative_signals: int
    alert: Optional[str] = None


# =============================================================================
# MICRO-COACHING
# =============================================================================

class MicroCoachingRequest(BaseModel):
    """Request für Micro-Coaching Feedback"""
    action_type: Literal["message_sent", "response_received", "deal_closed", "deal_lost"]
    context: Dict[str, Any] = Field(default_factory=dict)


class MicroCoachingResponse(BaseModel):
    """Response mit Micro-Coaching Feedback"""
    feedback: str
    feedback_type: Literal["positive", "tip", "warning"]


# =============================================================================
# PHONE MODE
# =============================================================================

class PhoneModeStartRequest(BaseModel):
    """Request zum Starten des Phone Mode"""
    lead_id: str
    lead_name: str
    call_type: Literal["discovery", "pitch", "close", "follow_up"]


class PhoneModeCoachingResponse(BaseModel):
    """Live-Coaching Response während Phone Call"""
    tag: str  # z.B. "[ÖFFNER]", "[EINWAND]", "[SIGNAL]"
    coaching: str
    urgency: Literal["low", "medium", "high"]


# =============================================================================
# COMPETITIVE INTELLIGENCE
# =============================================================================

class CompetitorMentionRequest(BaseModel):
    """Request wenn Wettbewerber erwähnt wird"""
    competitor_name: str
    mention_type: Literal["lock", "price", "feature", "third_party"]
    context: Optional[str] = None


class CompetitorHandlingResponse(BaseModel):
    """Response für Wettbewerber-Handling"""
    competitor_name: str
    strategy: str
    response_template: str
    do: List[str]
    dont: List[str]


# =============================================================================
# A/B TESTING
# =============================================================================

class ABTestCreateRequest(BaseModel):
    """Request zum Erstellen eines A/B Tests"""
    name: str
    description: Optional[str] = None
    test_type: Literal["framework", "industry", "buyer_type", "language"]
    variant_a: str
    variant_b: str
    target_metric: str = Field(default="conversion_rate")
    target_industry: Optional[IndustryType] = None
    target_buyer_type: Optional[BuyerType] = None


class ABTestResponse(BaseModel):
    """Response für A/B Test"""
    id: str
    name: str
    test_type: str
    variant_a: str
    variant_b: str
    variant_a_count: int
    variant_b_count: int
    variant_a_conversions: int
    variant_b_conversions: int
    variant_a_rate: float
    variant_b_rate: float
    winner: Optional[str] = None
    statistical_significance: float
    status: Literal["running", "completed", "paused"]
    created_at: datetime
    updated_at: datetime


class ABTestResultRequest(BaseModel):
    """Request zum Loggen eines A/B Test Ergebnisses"""
    test_id: str
    variant: Literal["a", "b"]
    converted: bool
    lead_id: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None


# =============================================================================
# ANALYTICS - FRAMEWORK EFFECTIVENESS
# =============================================================================

class FrameworkEffectivenessRequest(BaseModel):
    """Request für Framework-Effectiveness Analyse"""
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    industry_filter: Optional[IndustryType] = None
    buyer_type_filter: Optional[BuyerType] = None


class FrameworkEffectivenessItem(BaseModel):
    """Effectiveness eines Frameworks"""
    framework_id: str
    framework_name: str
    total_uses: int
    conversions: int
    conversion_rate: float
    avg_deal_value: Optional[float] = None
    avg_time_to_close_days: Optional[float] = None
    best_for_buyer_types: List[str]
    best_for_industries: List[str]


class FrameworkEffectivenessResponse(BaseModel):
    """Response für Framework-Effectiveness"""
    period_start: datetime
    period_end: datetime
    total_deals: int
    frameworks: List[FrameworkEffectivenessItem]
    top_framework: str
    insights: List[str]


class BuyerTypeEffectivenessItem(BaseModel):
    """Effectiveness nach Buyer Type"""
    buyer_type: BuyerType
    total_leads: int
    conversions: int
    conversion_rate: float
    best_framework: str
    avg_touchpoints: float


class BuyerTypeEffectivenessResponse(BaseModel):
    """Response für Buyer Type Effectiveness"""
    period_start: datetime
    period_end: datetime
    buyer_types: List[BuyerTypeEffectivenessItem]
    insights: List[str]


class IndustryEffectivenessItem(BaseModel):
    """Effectiveness nach Industry"""
    industry_id: str
    industry_name: str
    total_deals: int
    conversions: int
    conversion_rate: float
    best_framework: str
    best_buyer_approach: str
    avg_deal_size: float


class IndustryEffectivenessResponse(BaseModel):
    """Response für Industry Effectiveness"""
    period_start: datetime
    period_end: datetime
    industries: List[IndustryEffectivenessItem]
    insights: List[str]

File: salesflow-app/src/types/salesIntelligence.ts
/**
 * ╔════════════════════════════════════════════════════════════════════════════╗
 * ║  SALES INTELLIGENCE TYPES v3.0                                             ║
 * ║  TypeScript Types für Multi-Language, Buyer Psychology, Frameworks, etc.  ║
 * ╚════════════════════════════════════════════════════════════════════════════╝
 */

// =============================================================================
// ENUMS
// =============================================================================

export enum LanguageCode {
  DE = 'de',
  DE_AT = 'de-at',
  DE_CH = 'de-ch',
  EN_US = 'en-us',
  EN_UK = 'en-uk',
  ES = 'es',
  ES_LATAM = 'es-latam',
  FR = 'fr',
  IT = 'it',
  PT = 'pt',
  NL = 'nl',
  PL = 'pl',
  TR = 'tr',
}

export enum BuyerType {
  ANALYTICAL = 'analytical',
  DRIVER = 'driver',
  EXPRESSIVE = 'expressive',
  AMIABLE = 'amiable',
}

export enum BuyingStage {
  AWARENESS = 'awareness',
  CONSIDERATION = 'consideration',
  DECISION = 'decision',
  VALIDATION = 'validation',
}

export enum RiskProfile {
  RISK_AVERSE = 'risk_averse',
  RISK_NEUTRAL = 'risk_neutral',
  RISK_TAKER = 'risk_taker',
}

export enum AuthorityLevel {
  DECISION_MAKER = 'decision_maker',
  INFLUENCER = 'influencer',
  GATEKEEPER = 'gatekeeper',
  CHAMPION = 'champion',
  USER = 'user',
}

export enum FrameworkType {
  SPIN = 'spin',
  CHALLENGER = 'challenger',
  GAP = 'gap',
  SANDLER = 'sandler',
  SNAP = 'snap',
  MEDDIC = 'meddic',
  SOLUTION = 'solution',
}

export enum IndustryType {
  NETWORK_MARKETING = 'network_marketing',
  REAL_ESTATE = 'real_estate',
  INSURANCE = 'insurance',
  FINANCE = 'finance',
  B2B_SAAS = 'b2b_saas',
  B2B_SERVICES = 'b2b_services',
  COACHING = 'coaching',
  AUTOMOTIVE = 'automotive',
  RECRUITING = 'recruiting',
  HEALTHCARE = 'healthcare',
  EVENT_SALES = 'event_sales',
  RETAIL_HIGH_TICKET = 'retail_high_ticket',
}

export enum MomentumTrend {
  IMPROVING = 'improving',
  STABLE = 'stable',
  DECLINING = 'declining',
}

// =============================================================================
// LABELS & DISPLAY HELPERS
// =============================================================================

export const LANGUAGE_LABELS: Record<LanguageCode, string> = {
  [LanguageCode.DE]: '🇩🇪 Deutsch',
  [LanguageCode.DE_AT]: '🇦🇹 Österreichisch',
  [LanguageCode.DE_CH]: '🇨🇭 Schweizerdeutsch',
  [LanguageCode.EN_US]: '🇺🇸 US English',
  [LanguageCode.EN_UK]: '🇬🇧 British English',
  [LanguageCode.ES]: '🇪🇸 Español',
  [LanguageCode.ES_LATAM]: '🌎 Español (Latam)',
  [LanguageCode.FR]: '🇫🇷 Français',
  [LanguageCode.IT]: '🇮🇹 Italiano',
  [LanguageCode.PT]: '🇵🇹 Português',
  [LanguageCode.NL]: '🇳🇱 Nederlands',
  [LanguageCode.PL]: '🇵🇱 Polski',
  [LanguageCode.TR]: '🇹🇷 Türkçe',
};

export const BUYER_TYPE_LABELS: Record<BuyerType, string> = {
  [BuyerType.ANALYTICAL]: '🧮 Analytiker',
  [BuyerType.DRIVER]: '🎯 Macher',
  [BuyerType.EXPRESSIVE]: '✨ Visionär',
  [BuyerType.AMIABLE]: '🤝 Beziehungsmensch',
};

export const BUYER_TYPE_COLORS: Record<BuyerType, string> = {
  [BuyerType.ANALYTICAL]: '#3B82F6', // Blue
  [BuyerType.DRIVER]: '#EF4444',     // Red
  [BuyerType.EXPRESSIVE]: '#F59E0B', // Amber
  [BuyerType.AMIABLE]: '#10B981',    // Green
};

export const BUYING_STAGE_LABELS: Record<BuyingStage, string> = {
  [BuyingStage.AWARENESS]: '👀 Awareness',
  [BuyingStage.CONSIDERATION]: '🤔 Consideration',
  [BuyingStage.DECISION]: '✅ Decision',
  [BuyingStage.VALIDATION]: '🎉 Validation',
};

export const FRAMEWORK_LABELS: Record<FrameworkType, string> = {
  [FrameworkType.SPIN]: 'SPIN Selling',
  [FrameworkType.CHALLENGER]: 'Challenger Sale',
  [FrameworkType.GAP]: 'GAP Selling',
  [FrameworkType.SANDLER]: 'Sandler System',
  [FrameworkType.SNAP]: 'SNAP Selling',
  [FrameworkType.MEDDIC]: 'MEDDIC',
  [FrameworkType.SOLUTION]: 'Solution Selling',
};

export const INDUSTRY_LABELS: Record<IndustryType, string> = {
  [IndustryType.NETWORK_MARKETING]: 'Network Marketing',
  [IndustryType.REAL_ESTATE]: 'Immobilien',
  [IndustryType.INSURANCE]: 'Versicherungen',
  [IndustryType.FINANCE]: 'Finanzdienstleistungen',
  [IndustryType.B2B_SAAS]: 'B2B SaaS',
  [IndustryType.B2B_SERVICES]: 'B2B Services',
  [IndustryType.COACHING]: 'Coaching & Training',
  [IndustryType.AUTOMOTIVE]: 'Automotive',
  [IndustryType.RECRUITING]: 'Recruiting',
  [IndustryType.HEALTHCARE]: 'Healthcare',
  [IndustryType.EVENT_SALES]: 'Event Sales',
  [IndustryType.RETAIL_HIGH_TICKET]: 'High-Ticket Retail',
};

export const MOMENTUM_TREND_LABELS: Record<MomentumTrend, string> = {
  [MomentumTrend.IMPROVING]: '📈 Steigend',
  [MomentumTrend.STABLE]: '➡️ Stabil',
  [MomentumTrend.DECLINING]: '📉 Fallend',
};

// =============================================================================
// LANGUAGE TYPES
// =============================================================================

export interface CulturalProfile {
  language_code: string;
  language_name: string;
  formality_default: string;
  directness: string;
  relationship_first: boolean;
  urgency_acceptable: boolean;
  small_talk_expected: boolean;
  emoji_tolerance: string;
  typical_objection_style: string;
  trust_building_approach: string;
  closing_style: string;
}

export interface LanguageDetectionResult {
  language_code: string;
  regional_variant?: string;
  formality_detected: string;
  dialect_markers: string[];
  confidence: number;
}

export interface LocalizedTemplate {
  template_key: string;
  language_code: string;
  template_text: string;
}

// =============================================================================
// BUYER PSYCHOLOGY TYPES
// =============================================================================

export interface BuyerProfileAnalysis {
  buyer_type: BuyerType;
  buyer_type_confidence: number;
  buyer_type_signals: string[];
  
  buying_stage: BuyingStage;
  buying_stage_confidence: number;
  buying_stage_signals: string[];
  
  risk_profile: RiskProfile;
  risk_profile_confidence: number;
  risk_profile_signals: string[];
  
  authority_level: AuthorityLevel;
  authority_level_confidence: number;
  authority_level_signals: string[];
}

export interface CommunicationRecommendation {
  tone: string;
  message_length: string;
  emphasis: string[];
  avoid: string[];
  ideal_next_message: string;
}

export interface BuyerTypeInfo {
  buyer_type: BuyerType;
  name: string;
  emoji: string;
  characteristics: string[];
  communication_do: string[];
  communication_dont: string[];
  ideal_pitch: string;
  objection_handling: string;
  closing_approach: string;
  typical_questions: string[];
}

export interface ObjectionByBuyerType {
  buyer_type: BuyerType;
  objection_type: string;
  strategy: string;
  example: string;
}

// =============================================================================
// SALES FRAMEWORK TYPES
// =============================================================================

export interface FrameworkStage {
  name: string;
  goal: string;
  description: string;
  example_questions?: string[];
  example_approach?: string;
}

export interface SalesFramework {
  id: string;
  name: string;
  best_for: string[];
  core_principle: string;
  stages: FrameworkStage[];
  key_questions: string[];
  common_mistakes: string[];
}

export interface FrameworkRecommendation {
  recommended_framework: FrameworkType;
  reasoning: string;
  alternatives: FrameworkType[];
}

export interface ObjectionByFramework {
  framework_id: string;
  objection_type: string;
  response: string;
}

// =============================================================================
// INDUSTRY TYPES
// =============================================================================

export interface BuyerPersona {
  name: string;
  desc: string;
}

export interface IndustryProfile {
  id: string;
  name: string;
  description: string;
  typical_sales_cycle: string;
  avg_deal_size: string;
  key_decision_factors: string[];
  typical_objections: string[];
  compliance_rules: string[];
  recommended_frameworks: string[];
  buyer_personas: BuyerPersona[];
  communication_style: {
    tone: string;
    channel: string;
    frequency: string;
  };
  trust_builders: string[];
  red_flags: string[];
}

export interface IndustryObjection {
  industry_id: string;
  objection_type: string;
  strategy: string;
  example: string;
}

// =============================================================================
// MOMENTUM TYPES
// =============================================================================

export interface MomentumSignal {
  type: 'positive' | 'negative' | 'neutral';
  signal: string;
  weight: number;
  description: string;
}

export interface MomentumScore {
  lead_id: string;
  score: number;
  trend: MomentumTrend;
  recommendation: string;
  signals_count: number;
  positive_signals: number;
  negative_signals: number;
  alert?: string;
}

// =============================================================================
// MICRO-COACHING TYPES
// =============================================================================

export interface MicroCoachingFeedback {
  feedback: string;
  feedback_type: 'positive' | 'tip' | 'warning';
}

// =============================================================================
// PHONE MODE TYPES
// =============================================================================

export interface PhoneModeSession {
  session_id: string;
  lead_id: string;
  lead_name: string;
  call_type: 'discovery' | 'pitch' | 'close' | 'follow_up';
  status: 'active' | 'ended';
}

export interface PhoneModeCoaching {
  tag: string;
  coaching: string;
  urgency: 'low' | 'medium' | 'high';
}

// =============================================================================
// COMPETITIVE INTELLIGENCE TYPES
// =============================================================================

export interface CompetitorHandling {
  competitor_name: string;
  strategy: string;
  response_template: string;
  do: string[];
  dont: string[];
}

// =============================================================================
// A/B TESTING TYPES
// =============================================================================

export interface ABTest {
  id: string;
  name: string;
  test_type: 'framework' | 'industry' | 'buyer_type' | 'language';
  variant_a: string;
  variant_b: string;
  variant_a_count: number;
  variant_b_count: number;
  variant_a_conversions: number;
  variant_b_conversions: number;
  variant_a_rate: number;
  variant_b_rate: number;
  winner?: 'a' | 'b';
  statistical_significance: number;
  status: 'running' | 'completed' | 'paused';
  created_at: string;
  updated_at: string;
}

export interface ABTestCreateRequest {
  name: string;
  description?: string;
  test_type: 'framework' | 'industry' | 'buyer_type' | 'language';
  variant_a: string;
  variant_b: string;
  target_metric?: string;
  target_industry?: IndustryType;
  target_buyer_type?: BuyerType;
}

// =============================================================================
// ANALYTICS TYPES
// =============================================================================

export interface FrameworkEffectiveness {
  framework_id: string;
  framework_name: string;
  total_uses: number;
  conversions: number;
  conversion_rate: number;
  avg_deal_value?: number;
  avg_time_to_close_days?: number;
  best_for_buyer_types: string[];
  best_for_industries: string[];
}

export interface FrameworkEffectivenessReport {
  period_start: string;
  period_end: string;
  total_deals: number;
  frameworks: FrameworkEffectiveness[];
  top_framework: string;
  insights: string[];
}

export interface BuyerTypeEffectiveness {
  buyer_type: BuyerType;
  total_leads: number;
  conversions: number;
  conversion_rate: number;
  best_framework: string;
  avg_touchpoints: number;
}

export interface BuyerTypeEffectivenessReport {
  period_start: string;
  period_end: string;
  buyer_types: BuyerTypeEffectiveness[];
  insights: string[];
}

export interface IndustryEffectiveness {
  industry_id: string;
  industry_name: string;
  total_deals: number;
  conversions: number;
  conversion_rate: number;
  best_framework: string;
  best_buyer_approach: string;
  avg_deal_size: number;
}

export interface IndustryEffectivenessReport {
  period_start: string;
  period_end: string;
  industries: IndustryEffectiveness[];
  insights: string[];
}

// =============================================================================
// API REQUEST/RESPONSE HELPERS
// =============================================================================

export interface SalesIntelligenceState {
  selectedLanguage: LanguageCode;
  selectedIndustry: IndustryType;
  selectedFramework?: FrameworkType;
  detectedBuyerType?: BuyerType;
  detectedBuyingStage?: BuyingStage;
  momentumScore?: number;
  activeABTests: ABTest[];
}

// Default state
export const DEFAULT_SALES_INTELLIGENCE_STATE: SalesIntelligenceState = {
  selectedLanguage: LanguageCode.DE,
  selectedIndustry: IndustryType.NETWORK_MARKETING,
  activeABTests: [],
};

